File: HackerRank/ProjectEuler/euler005/euler005.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static int max(int a, int b)
    {
        if(a > b)   return a;
        return b;
    }
    
    static void Main(String[] args) {
        int t = Convert.ToInt32(Console.ReadLine());
        int[] if_prime;    if_prime = new int[50];
        for(int i = 0; i < 50; i++)
            if_prime[i] = 1;
        if_prime[0] = (if_prime[1] = 0);
        for(int i = 2; i < 50; i++)
            if(if_prime[i] == 1)
                for(int j = i*i; j < 50; j+=i)
                    if_prime[j] = 0;
        while(t != 0){
            t--;
            int[] MaxPower;
            MaxPower = new int[50];
            for(int i = 0; i < 50; i++) MaxPower[i] = 0;
            int n = Convert.ToInt32(Console.ReadLine());
            long res = 1;
            for(int i = 1; i <= n; i++){
                for(int j = 2; j <= i; j++){
                    int temp = i, power = 0;
                    while(if_prime[j] == 1 && temp % j == 0){
                        MaxPower[j] = max(MaxPower[j], ++power);
                        temp /= j;
                    }
                }
            }
            for(int j = 2; j <= n; j++){
                while(MaxPower[j] != 0){
                    res = res*j;
                    MaxPower[j]--;
                }
            }
            Console.WriteLine(res);
        }
    }
}

File: HackerRank/ProjectEuler/euler005/euler005.py
def max(a, b):
    if(a > b):
        return a
    return b

t = int(input())
prime = [None]*41
for i in range(41):
    prime[i] = 0

for i in range(2,41):
    if (prime[i] == 0):
        j = i*i
        while(j < 41):
            prime[j] = 1
            j += i
            
prime[0] = 1
prime[1] = 1

for tc in range(t):
    l = [None]*41
    for i in range(41):
        l[i] = 0
    n = int(input())
    for j in range(2,n+1):
        for i in range(2, j+1):
            power = 0
            temp = j
            while (prime[i] == 0 and temp % i == 0):
                power = power+1
                l[i] = max(l[i], power)
                temp /= i
                
    res = 1
    for i in range(2, n+1):
        while (l[i] != 0):
            res = res*i
            l[i] = l[i]-1
    print (res)
